packages/mongodb-constants/src/stage-operators.ts

```diff
@@ -0,0 +1,1014 @@
+import { ATLAS, ADL, ON_PREM } from './env';
+
+import {
+  ANY_NAMESPACE,
+  DATABASE,
+  ANY_COLLECTION_NAMESPACE,
+  COLLECTION,
+} from './ns';
+
+/**
+ * The stage operators.
+ */
+const STAGE_OPERATORS = [
+  {
+    name: '$addFields',
+    value: '$addFields',
+    label: '$addFields',
+    outputStage: false,
+    fullScan: false,
+    firstStage: false,
+    score: 1,
+    env: [ATLAS, ADL, ON_PREM],
+    meta: 'stage',
+    version: '3.4.0',
+    apiVersions: [1],
+    namespaces: [...ANY_NAMESPACE],
+    description:
+      'Adds new field(s) to a document with a computed value, or reassigns an existing field(s) with a computed value.',
+    comment: `/**
+ * newField: The new field name.
+ * expression: The new field expression.
+ */
+`,
+    snippet: `{
+  \${1:newField}: \${2:expression}, \${3:...}
+}`,
+  },
+  {
+    name: '$bucket',
+    value: '$bucket',
+    label: '$bucket',
+    outputStage: false,
+    fullScan: true,
+    firstStage: false,
+    score: 1,
+    env: [ATLAS, ADL, ON_PREM],
+    meta: 'stage',
+    version: '3.4.0',
+    apiVersions: [1],
+    namespaces: [...ANY_NAMESPACE],
+    description:
+      'Categorizes incoming documents into groups, called buckets, based on specified boundaries.',
+    comment: `/**
+ * groupBy: The expression to group by.
+ * boundaries: An array of the lower boundaries for each bucket.
+ * default: The bucket name for documents that do not fall within the specified boundaries
+ * output: {
+ *   outputN: Optional. The output object may contain a single or numerous field names used to accumulate values per bucket.
+ * }
+ */
+`,
+    snippet: `{
+  groupBy: \${1:expression},
+  boundaries: [ \${2:lowerbound}, \${3:...} ],
+  default: \${4:literal},
+  output: {
+    \${5:outputN}: { \${6:accumulator} }, \${7:...}
+  }
+}`,
+  },
+  {
+    name: '$bucketAuto',
+    value: '$bucketAuto',
+    label: '$bucketAuto',
+    outputStage: false,
+    fullScan: true,
+    firstStage: false,
+    score: 1,
+    env: [ATLAS, ADL, ON_PREM],
+    meta: 'stage',
+    version: '3.4.0',
+    apiVersions: [1],
+    namespaces: [...ANY_NAMESPACE],
+    description:
+      'Automatically categorizes documents into a specified number of buckets, attempting even distribution if possible.',
+    comment: `/**
```

Comment:
not sure if its important: once we have a bug in compass where stage comment was not valid and it required to update this package. maybe we can add test to validate if the comment is valid or not?

Reply:
Can you link to the case? This sounds like a reasonable thing to add, but I am not sure what valid comment would be / what to validate for here 🤔

Reply:
Aha, gotcha, thank you!

Reply:
Added a test https://github.com/mongodb-js/devtools-shared/pull/34/commits/f21f3d1676357f7497592c18f04462fd51064bfb

Does this make sense to you?

Reply:
yes, that suffices!